### PR TITLE
Add anthos deployment demo

### PR DESCRIPTION
**Pull request by abelgana**
This PR adds a deployment of Anthos.

### CODE REVIEW

File: anthos-deployment/.gitignore [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+/Godeps/
+
+### macOS ###
+# General
+.DS_Store
+.AppleDouble
+.LSOverride
+
+# Icon must end with two \r
+Icon
```
Comment (archyufa) [inline anthos-deployment/.gitignore]: remove space

File: anthos-deployment/main.tf [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  depends_on = [google_project_service.container]
+}
+
+resource "google_container_node_pool" "primary_preemptible_nodes" {
+  provider = google-beta
+
+  name       = var.node_name
+  location   = var.location
+  cluster    = google_container_cluster.primary.name
+  node_count = 4
```
Comment (archyufa) [inline anthos-deployment/main.tf]: Shall we use var here ?

**Review by archyufa — COMMENTED** [review]
Can you please add better description what in this PR getting installed

File: anthos-deployment/main.tf [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  node_config {
+    preemptible  = true
```
Comment (mathieu-benoit) [inline anthos-deployment/main.tf]: Could we add a variable for this?

File: anthos-deployment/main.tf [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+  release_channel {
+    channel = var.release_channel
+  }
+
+  depends_on = [google_project_service.container]
+}
+
+resource "google_container_node_pool" "primary_preemptible_nodes" {
+  provider = google-beta
```
Comment (mathieu-benoit) [inline anthos-deployment/main.tf:17]: Just curious, which GKE's feature require the `beta` version?

File: anthos-deployment/main.tf [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+module "acm" {
+  source = "terraform-google-modules/kubernetes-engine/google//modules/acm"
+
+  project_id       = var.default_project
+  cluster_name     = var.primary_name
+  location         = var.location
+  cluster_endpoint = google_container_cluster.primary.endpoint
+
+  sync_repo   = "git@github.com:abelgana/csp-config-management.git"
```
Comment (mathieu-benoit) [inline anthos-deployment/main.tf]: Should we add a variable here?

File: anthos-deployment/provider.tf [hunk 6]
```diff
@@ -0,0 +1,33 @@
+provider "google" {
+  version = "~> 3.49.0"
+  alias   = "tokengen"
+}
+data "google_client_config" "default" {
+  provider = google.tokengen
+}
+data "google_service_account_access_token" "sa" {
+  provider               = google.tokengen
+  target_service_account = "terraform@apt-rope-287612.iam.gserviceaccount.com"
```
Comment (mathieu-benoit) [inline anthos-deployment/provider.tf]: Should we add a variable here?

**Review by mathieu-benoit — COMMENTED** [review]
Great job! Few questions/comments + I'm wondering if it could be ideal to have in the readme file a section with the associated `terraform init`, `terraform plan` and `terraform apply` commands?

File: anthos-deployment/main.tf (hunk 4 above)
Comment (abelgana) [reply anthos-deployment/main.tf:17]: I had something,  don't recall but I removed it now.

File: anthos-deployment/main.tf [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  initial_node_count       = 1
+
+  release_channel {
+    channel = var.release_channel
+  }
+
+  depends_on = [google_project_service.container]
+}
+
+resource "google_container_node_pool" "primary_preemptible_nodes" {
```
Comment (mathieu-benoit) [inline anthos-deployment/main.tf:16]: For a learning perspective, any reason why here we are not using the GKE TF module from `terraform-google-modules/kubernetes-engine`?

Comment (archyufa) [reply anthos-deployment/main.tf:16]: I don't think our goal is to have production deployment more demo environment. We thought to use CFT initially though. The main issues is that if you want to addd something to CFT you need to make PR towards CFT and good luck with that :)

**Comment by archyufa** [comment]
lgtm

File: anthos-deployment/main.tf (hunk 7 above)
Comment (mathieu-benoit) [reply anthos-deployment/main.tf:16]: My point is more around, for the other resources we are using a module but not for the GKE one, was just curious to know why ;)

**Review by cartyc — APPROVED** [review]

File: anthos-deployment/main.tf (hunk 7 above)
Comment (abelgana) [reply anthos-deployment/main.tf:16]: I think the next iteration is to use a module. I was just hacking a demo first. I will open a task.

**Comment by abelgana** [comment]
@GCP-AppMod-Canada/contributors  I will merge this in an hour. Is that ok?

**Comment by mathieu-benoit** [comment]
/lgtm - assuming there will be at some point a snippet to run this?
```
gcloud ...
terraform init ...
terraform plan ...
terraform apply ...
```

**Review by mathieu-benoit — APPROVED** [review]

**Review by cartyc — APPROVED** [review]

**Pull request merged**